Add basic paging

samgensburg-gov opened the pull request:
Ended up doing a little more cleanup before actually getting into feature work.

aviars commented:
@samgensburg-gov As per the FHIR spec, read, update, delete share the same URL pattern and only differ by the HTTP verb. The  functionality inspects the HTTP action to determine whether the incoming request is actually for reading.  This removes the ability to add update/delete functionality. The PUT/DELETE verbs should return an operational error instead of removing.  It doesn't look like this PR adds paging.  It only removes functionality.  Its hard to asses the impact of this PR because my FHIR API tests are failing.
....
    apps/fhir/bluebutton/views/search.py", line 430, in read_search
        return JsonResponse(od['bundle'])
   TypeError: In order to allow non-dict objects to be serialized set the safe parameter to False

samgensburg-gov commented:
@aviars I fixed the one remaining integration test issue, so this should be ready to merge.

I removed any reference to update and delete functionality because Blue Button is only a read API. There's essentially no chance we will ever make this a writeable system, and even if we did, it would be such a massive architecture change that this small change here would not substantially impact that effort.

PUT and DELETE now return 405 errors, which is the correct response.
